### PR TITLE
Kinesis input compressed metrics

### PR DESCRIPTION
**Pull request by morfien101**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

###############################################

This code change is a supplement to PR #5588

With this you can now download the compressed metrics using the Telegraf Kinesis Input plugin.

### CODE REVIEW

File: plugins/inputs/kinesis_consumer/kinesis_consumer.go [hunk 1]
```diff
@@ -39,6 +44,8 @@ type (
 		ShardIteratorType      string    `toml:"shard_iterator_type"`
 		DynamoDB               *DynamoDB `toml:"checkpoint_dynamodb"`
 		MaxUndeliveredMessages int       `toml:"max_undelivered_messages"`
+		CompressedMetrics      bool      `toml:"compressed_metrics"`
```
Comment (glinton) [inline plugins/inputs/kinesis_consumer/kinesis_consumer.go]: Let's remove this `bool` and use the `compression_type` only.

File: plugins/inputs/kinesis_consumer/kinesis_consumer.go [hunk 2]
```diff
@@ -237,7 +251,42 @@ func (k *KinesisConsumer) Start(ac telegraf.Accumulator) error {
...
+		decompressedData, err = decompressSnappy(input)
+		if err != nil {
+			return nil, err
+		}
+	case compressionGZIP:
+		decompressedData, err = decompressGZip(input)
+		if err != nil {
+			return nil, err
+		}
+	default:
```
Comment (glinton) [inline plugins/inputs/kinesis_consumer/kinesis_consumer.go]: Default to uncompressed and set `decompressedData = input`; add a case for a compression type of "none" and do the same.

Comment (morfien101) [reply plugins/inputs/kinesis_consumer/kinesis_consumer.go]: I'm happy to do this. Can we do the same for the output plugin in that case. #5588

**Comment by sjwang90** [comment]
Closing this since we added this as an external plugin to #5588. Feel free to re-open if that's not the case.

**Pull request closed without merging**